src/yunohost/user.py

```diff
@@ -177,14 +177,17 @@ def user_create(auth, username, firstname, lastname, mail, password,
         try:
             with open('/etc/ssowat/conf.json.persistent') as json_conf:
                 ssowat_conf = json.loads(str(json_conf.read()))
+        except ValueError:
+            raise MoulinetteError(errno.EINVAL,
+                                  m18n.n('ssowat_persistent_conf_error'))
+        except IOError:
+            ssowat_conf = {}
 
-            if 'redirected_urls' in ssowat_conf and '/' in ssowat_conf['redirected_urls']:
-                del ssowat_conf['redirected_urls']['/']
-
-            with open('/etc/ssowat/conf.json.persistent', 'w+') as f:
-                json.dump(ssowat_conf, f, sort_keys=True, indent=4)
+        if 'redirected_urls' in ssowat_conf and '/' in ssowat_conf['redirected_urls']:
+            del ssowat_conf['redirected_urls']['/']
 
-        except IOError: pass
+        with open('/etc/ssowat/conf.json.persistent', 'w+') as f:
+            json.dump(ssowat_conf, f, sort_keys=True, indent=4)
```

Comment:
Got it !
I guess we'd like to have to different error message then, for both *read* and *write* errors.